### DISCUSSION TITLE
NVFORTRAN-S-1254-pi is use associated with math and cannot be redeclared. (rot_cyl.f90: 23)

### DISCUSSION THREAD

**Discussion by XueZhengHe03** [ExtremeFLOW/neko#1091] [Q&A] [answered]
I use the  way as https://github.com/ExtremeFLOW/flettner_rotor  :
/path/to/neko_install/bin/makeneko rot_cyl.f90, but It has the following problem 

Building user NEKO ...rot_cyl.f90:
NVFORTRAN-S-1254-pi is use associated with math and cannot be redeclared. (rot_cyl.f90: 23)
  0 inform,   0 warnings,   1 severes, 0 fatal for trip
NVFORTRAN-S-0142-fluid_usr_ic is not a component of this OBJECT (rot_cyl.f90: 999)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 999)
NVFORTRAN-S-0142-fluid_usr_if is not a component of this OBJECT (rot_cyl.f90: 1000)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 1000)
NVFORTRAN-S-0142-fluid_usr_f_vector is not a component of this OBJECT (rot_cyl.f90: 1001)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 1001)
NVFORTRAN-S-0142-fluid_usr_f is not a component of this OBJECT (rot_cyl.f90: 1002)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 1002)
NVFORTRAN-S-0142-usr_chk is not a component of this OBJECT (rot_cyl.f90: 1004)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 1004)
NVFORTRAN-S-0142-usr_msh_setup is not a component of this OBJECT (rot_cyl.f90: 1005)
NVFORTRAN-S-0155-Illegal POINTER assignment - non-POINTER object (rot_cyl.f90: 1005)
  0 inform,   0 warnings,  12 severes, 0 fatal for user_setup

my version is: 
nvfortran 23.11-0 64-bit target on x86-64 Linux -tp cascadelake
NVIDIA Compilers and Tools
Copyright (c) 2023, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.

Could you please help me to figure out what's wrong with rot_cyl.f90
I will fully appreciate it.

**Comment by njansson** [comment, marked as answer]
The Flettner rotor example found in https://github.com/ExtremeFLOW/flettner_rotor is for an old, unsupported version of Neko. Major changes needs to be done for the user interface in order to make `rot_cyl.f90` compile with recent version of Neko.
